### ISSUE TITLE
Restore Checks on `dom_username`?

### ISSUE THREAD

**Issue by cycomachead** [bromagosa/snapCloud#29]
There used to be validations on dom_username, but they were removed:

https://github.com/bromagosa/snapCloud/commit/6990be17040d9407addeaf650e0f3dbe01c004be#diff-bef302d3cdd1ce8c0d4e2e008fa3652a

If there's no validations, then should we not just use the text type? (It would make writing the initial migrations in #10 Easier).

**Comment by cycomachead** [comment]
It could also be enforced with just a constraint on the users table.

Also, does it make sense to enforce the lowercase name at the DB level? We could just create an index on `lower(username)` for that to work.

**Comment by bromagosa** [comment]
Yes, I removed those checks because MioSoft didn't have them, and some usernames (like bh) couldn't be imported otherwise. I'll reactivate them once the final migration is done.

I think it makes sense to enforce every policy everywhere: frontend, backend and db. Let's revisit this after the migration too.

**Comment by cycomachead** [comment]
Came across this recently, which is interesting. 
https://www.b-list.org/weblog/2018/feb/11/usernames/

It makes me think that we should think a little more broadly about this issue.

**Comment by cycomachead** [comment]
https://snap-analytics.herokuapp.com/question/17
